src/drivers/usb/class/usb_mass_storage.c

```diff
@@ -152,40 +125,47 @@ static void usb_mass_handle(struct usb_class *cls, struct usb_dev *dev) {
 		}
 	}
 
-#if 0
-	usb_endp_control(dev->endpoints[0], usb_mass_reset_hnd, NULL,
-			USB_DEV_REQ_TYPE_WR | USB_DEV_REQ_TYPE_CLS
-				| USB_DEV_REQ_TYPE_IFC,
-			USB_REQ_MASS_RESET, 0,
-			dev->iface_desc.b_interface_number, 0, NULL);
-#endif
 	usb_endp_control(dev->endpoints[0], usb_mass_maxlun_hnd, NULL,
-			USB_DEV_REQ_TYPE_RD | USB_DEV_REQ_TYPE_CLS
-				| USB_DEV_REQ_TYPE_IFC,
+			USB_DIR_IN | USB_REQ_TYPE_CLASS | USB_REQ_RECIP_IFACE,
 			USB_REQ_MASS_MAXLUN, 0,
 			dev->iface_desc.b_interface_number, 1,
 			&mass->maxlun);
 }
 
-static void usb_mass_release(struct usb_class *cls, struct usb_dev *dev) {
-	struct usb_mass *mass = usb2massdata(dev);
+static int usb_ms_probe(struct usb_dev *dev) {
+	struct usb_mass *mass = pool_alloc(&usb_mass_classes);
 
-	scsi_dev_detached(&mass->scsi_dev);
+	mass = pool_alloc(&usb_mass_classes);
```

Comment:
Double pool allocation (here and 2 lines upper)

Reply:
Thank you, fixed.